[AN] bug: 북마크 시 오디오 초기화되는 문제 수정 #226

rosemin928 opened the pull request:
<!-- PR 제목 예시
ex. [BE] 로그인 API 구현
ex. [AN] 페이지 구현 -->

## 📌 변경 내용 & 이유
<!-- ex. 
- 사용자 로그인 기능 구현
- 로그인 실패 시 에러 메시지 반환 처리 추가 -->
- 탐색 화면 북마크 시 오디오 초기화되는 문제 수정
- 탐색 화면 -> 상세 화면에서 북마크 -> 탐색 화면으로 돌아왔을 때 북마크 갱신 안 되는 문제 수정
- 상세 화면 viewModel 리팩터링

## 📸 스크린샷 (선택)
> UI 변경이 있을 경우 첨부

## 🧪 테스트 방법 (선택)
<!-- 코드 리뷰하는 팀원이 돌려볼 수 있도록 작성해주기 
ex. ReservationTest의 XXXX 테스트메서드로 확인 
ex. 어떤화면의 어떤기능에서 확인 -->

## 📢 논의하고 싶은 내용
<!-- 이러이러해서 이부분을 이러이러하게 구현했는데 이방식이 괜찮은지? -->

## 🧩 관련 이슈
<!-- 이슈 태그: #123 -->
<!-- 이슈 닫기: close #123 -->
#226 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 플레이어 상세 화면에서 이전 화면이 "탐색"일 경우 북마크 정보를 반영하여 돌아갈 수 있도록 개선되었습니다.

* **버그 수정**
  * 북마크 상태가 플레이어 상세 화면과 탐색 화면 간에 정확하게 동기화되도록 수정되었습니다.
  * 북마크 버튼의 선택 상태가 실제 북마크 ID 존재 여부에 따라 올바르게 표시됩니다.

* **UI/UX 개선**
  * 북마크 버튼 클릭 시 즉시 시각적 피드백이 제공됩니다.
  * 탐색 화면에서 북마크 상태 변경 시, 리스트에 바로 반영됩니다.

* **기타**
  * 사용되지 않는 두 개 활동 화면과 관련 레이아웃이 제거되어 앱 용량과 유지보수가 개선되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rosemin928 committed:
[AN] fix: 북마크했을 때 오디오 초기화되는 문제 수정 #226

rosemin928 committed:
[AN] refactor: 상세 화면 isBookmarked LiveData 제거 #226

HamBeomJoon commented on android/app/src/main/java/com/onair/hearit/presentation/detail/PlayerDetailViewModel.kt:
```diff
@@ -83,7 +78,6 @@ class PlayerDetailViewModel(
                     _keywords.value = it.keywords
-                    _isBookmarked.value = it.isBookmarked
                     _bookmarkId.value = it.bookmarkId
                 }.onFailure { it: Throwable ->
```
it: Throwable을 사용중인건가요?

rosemin928 replied:
안 쓰고 있네요 😅
삭제했습니다!

HamBeomJoon commented on android/app/src/main/java/com/onair/hearit/presentation/explore/ShortsViewHolder.kt:
```diff
@@ -45,6 +45,14 @@ class ShortsViewHolder(
 
+        binding.btnExploreItemBookmark.setOnClickListener {
+            binding.btnExploreItemBookmark.isSelected =
+                binding.btnExploreItemBookmark.isSelected.not()
```
그동안 저희가 쓴 코드는 not()보다는 !를 써와서 !를 사용하는건 어떤가요?

rosemin928 replied:
앟 그렇네용 !로 수정했습니다!

rosemin928 committed:
[AN] fix: 탐색-상세(북마크 변경)-탐색 이동했을 때 북마크 갱신이 안 되는 문제 수정 #226

rosemin928 committed:
[AN] refactor: 사용하지 않는 파일 삭제 #226

gahyunkim reviewed (approved):
북마크 문제 해결하는데 수고 많으셨습니다~!

rosemin928 committed:
Merge branch 'develop-an' into bug/#226-explore-bookmark-fix

rosemin928 committed:
[AN] refactor: 둘러보기 화면 북마크 아이디 업데이트 로직 수정 #226

- 기존 북마크 아이디를 덮어쓰는 문제를 해결하기 위해 초기 로드 시에는 북마크 아이디를 새로 설정하고, 이후에는 기존 북마크 아이디에 새로운 아이템의 북마크 아이디를 추가하도록 수정
